### PR TITLE
S3 t3/adding login validation logic

### PR DESCRIPTION
**Pull request by catai9**
**Scrum Board Ticket**: S3 T3: Implement endpoint logic to validate user email and password

**Story:** Story 3 Login by email

**Description:** Updated the validation method to check the database when verifying the user and added logic to correctly validate different types of users. 

**Testing Criteria:** Tested code by running locally and sending requests to the `/api/login` endpoint for valid and invalid credentials of both student and worker user types. Also ensured that all automated tests in Jest passed and had 100% file coverage.

**How Reviewers can Test Code:** 
1. Run code locally (navigate inside the `server` folder and run `npm start`). Test the `/api/login` endpoint on localhost port 4000 using Postman to ensure that the response is as expected. Pass valid and invalid credentials of both student and worker user types and check that the response is appropriate. A valid worker credential is `email=joshuabrooks@gmail.com` and `password=j1234`. A valid student credential is `email=johndoe@gmail.com` and `password=a1234`.
2. Run the automated tests to make sure all tests pass and have 100% file coverage. To do this, navigate inside the `server` folder and run `npm test`. Please note that tests have not been setup for the front end so an error will occur if one attempts to run `npm test` in the root folder.

### CODE REVIEW

File: server/models/handlers/authenticate.js [hunk 1]
```diff
@@ -1,19 +1,22 @@
...
+    // Searches the worker table to see if it is valid worker credentials
+    const worker = await workerModel.getWorker(email, password);
+    if  (!_.isNil(worker)) return worker;
 
-    return user;
+    // Returns null if the credentials do not match a student nor a worker's credentials.
+    return null;
 };
 
 // Verify the access token for each private request to improve security.
```
Comment (catai9) [inline server/models/handlers/authenticate.js:22]: Please note that access token verification is out of the scope of this story and will be implemented in a future sprint.

Comment (sehersarin) [reply server/models/handlers/authenticate.js:22]: Great - thanks for clarifying!

**Comment by sehersarin** [comment]
Overall, PR looks good. I have pulled the code and ran locally. Achieved a test coverage of 100% thus, will be merging into the master branch.

**Review by sehersarin — APPROVED** [review]
Have verified this PR by pulling the code and running locally. All tests passed and the expected state is achieved. Will be merging code to master branch!

**Pull request merged**